app/models/submission.rb

```diff
@@ -188,33 +181,23 @@ def self.get_submission_by_group_and_assignment(group_n, ass_si)
     end
   end
 
-  def create_remark_result
-    remark_result = Result.new
-    results << remark_result
-    remark_result.marking_state = Result::MARKING_STATES[:unmarked]
-    remark_result.submission_id = self.id
-    remark_result.save
-    # link remark result id to submission - must be done after remark result is saved (so it has an id)
-    self.remark_result_id = remark_result.id
+  def make_remark_result
+    remark = self.create_remark_result(
+      marking_state: Result::MARKING_STATES[:unmarked],
+      submission_id: id)
     self.save
 
     # populate remark result with old marks
     original_result = get_original_result
 
-    old_extra_marks = original_result.extra_marks
-    old_extra_marks.each do |old_extra_mark|
-      remark_extra_mark = ExtraMark.new(old_extra_mark.attributes.merge(
-        {result_id: self.remark_result_id, created_at: Time.zone.now}))
-      remark_extra_mark.save(validate: false)
-      remark_result.extra_marks << remark_extra_mark
+    original_result.extra_marks.each do |extra_mark|
+      remark.extra_marks.create(extra_mark.attributes.merge(
+        { result: remark, created_at: Time.zone.now }))
     end
 
-    old_marks = original_result.marks
-    old_marks.each do |old_mark|
-      remark_mark = Mark.new(old_mark.attributes.merge(
-        {result_id: self.remark_result_id, created_at: Time.zone.now}))
-      remark_mark.save(validate: false)
-      remark_result.marks << remark_mark
+    original_result.marks.each do |mark|
+      remark_result.marks.create(mark.attributes.merge(
+        { result: remark, created_at: Time.zone.now }))
```

Comment:
Redundant curly braces around a hash parameter.<br>Indent the first parameter one step more than `mark.attributes.merge(`.